### PR TITLE
#207 rollup compaction simplified

### PR DESCRIPTION
**Pull request by alexeiakimov**
## Description

A simplified approach to multi-block files and the rollup compaction #207. It is not a full solution, but an initial attempt to use multi-block files for indexing and queries. Replication is out of scope.

### CODE REVIEW

File: core/src/main/scala/io/qbeast/core/model/IndexFileBuilder.scala [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+      this
+    }
+
+    /**
+     * Sets the element count.
+     *
+     * @param elemenCount the elemenCount to set
+     * @return this instance
+     */
+    def setElemenCount(elementCount: Long): BlockBuilder = {
```
Comment (Jiaweihu08) [inline core/src/main/scala/io/qbeast/core/model/IndexFileBuilder.scala:156]: typo here, setElementCount

File: src/main/scala/io/qbeast/spark/delta/writer/IndexFileWriter.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  trackers.foreach(_.newFile(output.path()))
+
+  /**
+   * Writes a given row, that should not have Qbeast specific columns.
+   *
+   * @param row the row to write
+   * @param cubeId the cube identifier
+   * @param weight the weight
+   */
+  def write(row: InternalRow, cubeId: CubeId, weigth: Weight): Unit = {
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/spark/delta/writer/IndexFileWriter.scala:42]: Typo here: weight

Comment (alexeiakimov) [reply src/main/scala/io/qbeast/spark/delta/writer/IndexFileWriter.scala]: Fixed, thank you.

File: core/src/main/scala/io/qbeast/core/model/IndexFileBuilder.scala (hunk 1 above)
Comment (alexeiakimov) [reply core/src/main/scala/io/qbeast/core/model/IndexFileBuilder.scala]: Fixed, thank you.

**Comment by alexeiakimov** [comment]
This PR is closed because of incorrect target branch. The correct PR is #225

**Pull request closed without merging**